### DISCUSSION TITLE
Add a capability to skip stages in testing for larger designs?

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#1525] [General]
I think it could be useful in regression testing to skip stages that take too long.

For instance #1452 would be a useful test, if detailed routing could be skipped. I do not know why global placement takes so long, but until that is fixed(either in OpenROAD or by modifying the design), placement too should be skipped.

mock-array used to take hours(I forget how long) and now it takes minutes. So it would be nice to start regression testing a design before it is fully optimized...

Thoughts?

**Comment by maliberty** [comment]
This would imply missing metrics in the results which, at least currently, would make it fail.   Without placement or routing I'm not sure what we would be testing.   I think it is better to work through such issues before making it part of the regression suite.  

We could merge such a design as off by default and in development.  Is there any open issue for gpl?

**Reply by oharboe** [reply]
Makes sense.

No issues yet. I have not studied the mock-aluregisterfile design and the stages.

I could file a feature request to speed it up?

**Reply by maliberty** [reply]
Sure though the more specific you can be the better.

**Reply by oharboe** [reply]
Not sure what to ask for except to make it faster ... For me global placement is a black box....

Run perf? a bit of suspend/resume debugging to see if I can glean something?

**Reply by maliberty** [reply]
Its enough to mention that is slow in global placement.

**Reply by oharboe** [reply]
@maliberty Here you go... #1527
